secure/net/src/lib.rs

```diff
@@ -351,4 +363,57 @@ mod test {
         assert_eq!(data1, result1);
         assert_eq!(data2, result2);
     }
+
+    #[test]
+    fn test_server_timeout() {
+        let server_port = utils::get_available_port();
+        let server_addr = SocketAddr::new(IpAddr::V4(Ipv4Addr::LOCALHOST), server_port);
+        let mut server = NetworkServer::new(server_addr);
+        let mut client = NetworkClient::new(server_addr);
+        let data1 = vec![0, 1, 2, 3];
+        let data2 = vec![4, 5, 6, 7];
+
+        // First client, success
+        client.write(&data1).unwrap();
+        let result1 = server.read().unwrap();
+        assert_eq!(data1, result1);
+
+        // Timedout
+        server.read().unwrap_err();
+
+        // New client, success, note the previous client connection is still active, the server is
+        // actively letting it go due to lack of activity.
+        let mut client2 = NetworkClient::new(server_addr);
+        client2.write(&data2).unwrap();
+        let result2 = server.read().unwrap();
+        assert_eq!(data2, result2);
+    }
+
+    #[test]
+    fn test_client_timeout() {
+        let server_port = utils::get_available_port();
+        let server_addr = SocketAddr::new(IpAddr::V4(Ipv4Addr::LOCALHOST), server_port);
+        let mut server = NetworkServer::new(server_addr);
+        let mut client = NetworkClient::new(server_addr);
+        let data1 = vec![0, 1, 2, 3];
+        let data2 = vec![4, 5, 6, 7];
+
+        // First server success
+        client.write(&data1).unwrap();
+        let result1 = server.read().unwrap();
+        assert_eq!(data1, result1);
+
+        // Timedout, it is hard to simulate a client receiving a write timeout
+        client.read().unwrap_err();
+
+        // Clean up old Server listener but keep the stream online. Start a new server, which will
+        // be the one the client now connects to.
+        server.listener = None;
+        let mut server2 = NetworkServer::new(server_addr);
```

Comment:
does it work with the same client/server after timeout?

Reply:
It should, will expand the test.